### PR TITLE
Add format support

### LINKED ISSUE #12: Support plain "key value" input files
Would be useful to support files of the following format:
```
key1 value1
key2 some value 2
key3 42
```
Interpreted as if it was this YAML (see the first link for details):
```yaml
key1: "value1"
key2: "some value 2"
key3: "42"
```
This is the format produced by "workplace status" commands - see [`--workspace_status_command program`](https://docs.bazel.build/versions/master/user-manual.html#flag--workspace_status_command). Key/value pairs then can be consumed by Bazel rules, typically for stamping (introducing controlled non-determinism into the build). Typical example is putting git commit sha into a docker image tag (see `stamp` in [`container_push`](https://github.com/bazelbuild/rules_docker#container_push-1) in rules_docker), etc.

I'd like to add stamping support to rjsone rule for bazel (https://github.com/atlassian/bazel-tools/issues/12) and I have two options:
- write a helper program that translates the stamping format into yaml/json so that rjsone, invoked from the rule, can consume it as an additional context.
- modify rjsone to support this format so that rjsone rule can just pass the file(s), produced by some workspace status command, as additional contexts.

This issue is about the later approach. It may be a new flag for files in this formats or some modifier to existing flags. WDYT?

### PR DESCRIPTION
**Pull request by wryun**
'bazel' is called kv. See new test. Fixes #12

### CODE REVIEW

**Comment by wryun** [comment]
Feel free to just check the test files for the usage example (and merge if you want to use this).

File: context.go [hunk 1]
```diff
@@ -12,161 +12,248 @@ import (
...
+		if err := yaml_ghodss.Unmarshal(data, &result); err != nil {
 			return nil, err
 		}
-	}
+		return result, nil
+	case textFormat:
+		return string(data), nil
+	case kvFormat:
+		// TODO unicode?
+		lines := strings.Split(string(data), "\n")
```
Comment (wryun) [inline context.go:180]: this is the kv format loading. Code is supposed to be simple, right?

File: context.go [hunk 2]
```diff
@@ -12,161 +12,248 @@ import (
...
+				continue
+			}
 
-	if cc.key != "" {
-		return map[string]interface{}{cc.key: finalResult}, nil
+			splitLine := strings.SplitN(line, " ", 2)
+			if len(splitLine) != 2 {
+				return nil, fmt.Errorf("line not in kv format: %q", line)
+			}
+			result[splitLine[0]] = splitLine[1]
```
Comment (ash2k) [inline context.go:191]: Last key wins for duplicate key.

**Review by ash2k — APPROVED** [review]

**Comment by ash2k** [comment]
The readme needs to be updated? I'll start working on stamping using this commit. Feel free to merge it if you want to fix readme in a separate PR.

**Comment by wryun** [comment]
Yes, I want to fix the README separately. It's still accurate (this is a new feature), and I want to see how it works first.

Also, thinking of changing how functions work due to this to try to make it more consistent, so, yeah.

File: context.go (hunk 2 above)
Comment (wryun) [reply context.go:191]: yes, I should have commented. Should it error out? I can change.

Comment (ash2k) [reply context.go:191]: Hm, I think it should probably log to stderr about this? Maybe on debug level (disabled by default?). There should be a way to discover what is happening and debug logging is what comes to mind here.

**Review by Dustin4444 — COMMENTED** [review]
#17

**Review by Dustin4444 — COMMENTED** [review]
#13

**Pull request merged**